bin/run-tests.sh

```
@@ -12,36 +12,80 @@
+# Refuse to start off Linux rather than fail somewhere in the middle. This harness is a Linux tool by
+# choice (see the HOST note in the header); on a BSD userland it would not fail cleanly, it would
+# mis-hash the sandbox, mis-read a suite list and report a verdict that looks real.
+if [[ $(uname -s) != Linux ]]; then
+	echo "run-tests.sh is Linux-only (bash 4+, GNU coreutils, GNU find); this host is $(uname -s)." >&2
+	echo "Run the suite in a container instead: ./bin/test-native-docker.sh" >&2
+	exit 2
+fi
+
 SCRIPT_DIR="$(cd "$(dirname "$0")" && pwd)"
 ROOT_DIR="$(cd "$SCRIPT_DIR/.." && pwd)"
 cd "$ROOT_DIR"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)